zanata-war/src/main/java/org/zanata/action/LoginAction.java

```diff
@@ -141,12 +141,12 @@ public String login() {
 
     private String continueToPreviousUrl() {
         try {
-            FacesNavigationUtil.redirect(FacesContext.getCurrentInstance(),
-                    userRedirect.getUrl());
-        } catch (IOException e) {
+            urlUtil.redirectToInternal(userRedirect.getUrl());
+            return "continue";
+        } catch (Exception e) {
+            // already logged by urlUtil.redirectToInternal.
             return "dashboard";
```

Comment:
Sorry, I just realised what a terrible idea this is. Now we could be swallowing all sorts of exceptions, eg ClassNotFoundException, without logging anything.